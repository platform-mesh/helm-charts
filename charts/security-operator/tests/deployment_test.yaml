suite: operator
chart:
  version: 1.0.0
  appVersion: 1.0.0
tests:
  - it: operator match the snapshot
    set:
      kubeconfigSecret: kcp-kubeconfig
      caSecret: ca-secret
      initializer:
        extraArgs:
          - --test=myval
          - --test2=myval
        kubeconfigSecret: kcp-initializer-kubeconfig
    asserts:
      - matchSnapshot: {}
  - it: operator match the snapshot w. sentry
    set:
      sentry:
        enabled: true
      externalSecrets:
        enabled: true
      extraEnvs:
        - name: TEST_ENV
          value: test
      extraVolumes:
        - name: kcp-kubeconfig
          secret:
            secretName: kcp-kubeconfig
      extraVolumeMounts:
        - name: kcp-kubeconfig
          mountPath: /etc/kcp-kubeconfig
          readOnly: true
    asserts:
      - matchSnapshot: {}
  - it: has all health checks
    template: deployment.yaml
    asserts:
      - equal:
          path: spec.template.spec.containers[0].livenessProbe.httpGet.path
          value: /healthz
      - equal:
          path: spec.template.spec.containers[0].readinessProbe.httpGet.path
          value: /readyz
      - equal:
          path: spec.template.spec.containers[0].startupProbe.httpGet.path
          value: /readyz

