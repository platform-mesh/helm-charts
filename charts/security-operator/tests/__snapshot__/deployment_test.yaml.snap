operator match the snapshot:
  1: |
    apiVersion: rbac.authorization.k8s.io/v1
    kind: ClusterRole
    metadata:
      name: RELEASE-NAME-security-operator
    rules:
      - apiGroups:
          - coordination.k8s.io
        resources:
          - leases
        verbs:
          - create
          - delete
          - get
          - list
          - patch
          - update
          - watch
      - apiGroups:
          - ""
        resources:
          - events
          - secrets
        verbs:
          - '*'
      - apiGroups:
          - helm.toolkit.fluxcd.io
        resources:
          - helmreleases
        verbs:
          - get
          - list
          - watch
          - create
          - delete
          - update
          - patch
      - apiGroups:
          - source.toolkit.fluxcd.io
        resources:
          - ocirepositories
          - helmreleases
        verbs:
          - get
          - list
          - create
          - update
          - patch
          - delete
          - watch
  2: |
    apiVersion: rbac.authorization.k8s.io/v1
    kind: ClusterRoleBinding
    metadata:
      name: RELEASE-NAME-security-operator
    roleRef:
      apiGroup: rbac.authorization.k8s.io
      kind: ClusterRole
      name: RELEASE-NAME-security-operator
    subjects:
      - kind: ServiceAccount
        name: RELEASE-NAME-security-operator
        namespace: NAMESPACE
  3: |
    apiVersion: v1
    data:
      coreModule.fga: |2-
        module core

        type user

        type role
          relations
            define assignee: [user,user:*]

        type core_platform-mesh_io_account
          relations
            define parent: [core_platform-mesh_io_account]

            define owner: [role#assignee] or owner from parent
            define member: [role#assignee] or owner or member from parent

            define get: member
            define update: member
            define patch: member
            define delete: owner

            define create_core_platform-mesh_io_accounts: member
            define list_core_platform-mesh_io_accounts: member
            define watch_core_platform-mesh_io_accounts: member

            # org and account specific
            define watch: member

            define create_core_platform-mesh_io_accountinfos: member
            define list_core_platform-mesh_io_accountinfos: member
            define watch_core_platform-mesh_io_accountinfos: member

            define create_apis_kcp_io_apibindings: owner
            define list__apis_kcp_io_apibindings: member
            define watch_apis_kcp_io_apibindings: member

            # IAM specific
            define manage_iam_roles: owner
            define get_iam_roles: member
            define get_iam_users: member

        type core_platform-mesh_io_accountinfo
          relations
            define parent: [core_platform-mesh_io_account]

            define member: member from parent
            define owner: owner from parent

            define get: member
            define watch: member

            # IAM specific
            define manage_iam_roles: owner
            define get_iam_roles: member
            define get_iam_users: member

        type apis_kcp_io_apibinding
          relations
            define parent: [core_platform-mesh_io_account]

            define member: member from parent
            define owner: owner from parent

            define get: member
            define watch: member
            define update: owner
            define patch: owner
            define delete: owner

            # IAM specific
            define manage_iam_roles: owner
            define get_iam_roles: member
            define get_iam_users: member
    kind: ConfigMap
    metadata:
      name: RELEASE-NAME-security-operator-core-module
  4: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      name: RELEASE-NAME-security-operator
      namespace: NAMESPACE
    spec:
      revisionHistoryLimit: 3
      selector:
        matchLabels:
          app: RELEASE-NAME-security-operator
      template:
        metadata:
          labels:
            app: RELEASE-NAME-security-operator
            control-plane: controller-manager
        spec:
          automountServiceAccountToken: true
          containers:
            - args:
                - fga
                - --invite-keycloak-password-file=/keycloak-credentials/secret
                - --leader-elect
                - --metrics-bind-address=:9090
                - --health-probe-bind-address=:8090
                - --log-level=warn
                - --region=local
                - --environment=local
                - --image-tag=1.0.0
                - --image-name="ghcr.io/platform-mesh/security-operator"
                - --shutdown-timeout=1m
                - --max-concurrent-reconciles=10
              env:
                - name: FGA_TARGET
                  value: openfga.platform-mesh-system.svc.cluster.local:8081
                - name: KCP_KUBECONFIG
                  value: /api-kubeconfig/kubeconfig
              image: ghcr.io/platform-mesh/security-operator:1.0.0
              imagePullPolicy: IfNotPresent
              livenessProbe:
                failureThreshold: 1
                httpGet:
                  path: /healthz
                  port: 8090
                periodSeconds: 10
              name: RELEASE-NAME-security-operator
              ports:
                - containerPort: 9090
                  name: metrics
                  protocol: TCP
                - containerPort: 8090
                  name: health-port
                  protocol: TCP
              readinessProbe:
                httpGet:
                  path: /readyz
                  port: 8090
                initialDelaySeconds: 5
                periodSeconds: 10
              resources:
                limits:
                  cpu: 260m
                  memory: 512Mi
                requests:
                  cpu: 150m
                  memory: 128Mi
              securityContext:
                readOnlyRootFilesystem: true
                runAsNonRoot: true
                seccompProfile:
                  type: RuntimeDefault
              startupProbe:
                failureThreshold: 30
                httpGet:
                  path: /readyz
                  port: 8090
                periodSeconds: 10
              volumeMounts:
                - mountPath: /api-kubeconfig
                  name: external-api-server
                - mountPath: /keycloak-credentials
                  name: keycloak-credentials
                  readOnly: true
          securityContext:
            runAsNonRoot: true
            seccompProfile:
              type: RuntimeDefault
          serviceAccountName: RELEASE-NAME-security-operator
          terminationGracePeriodSeconds: 10
          volumes:
            - name: external-api-server
              secret:
                secretName: kcp-kubeconfig
            - name: keycloak-credentials
              secret:
                secretName: keycloak-admin
  5: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      name: RELEASE-NAME-security-operator-generator
      namespace: NAMESPACE
    spec:
      revisionHistoryLimit: 3
      selector:
        matchLabels:
          service: RELEASE-NAME-security-operator-generator
      template:
        metadata:
          labels:
            control-plane: controller-manager
            service: RELEASE-NAME-security-operator-generator
        spec:
          automountServiceAccountToken: true
          containers:
            - args:
                - model-generator
                - --leader-elect
                - --metrics-bind-address=:9090
                - --health-probe-bind-address=:8090
                - --log-level=warn
                - --region=local
                - --environment=local
                - --image-tag=1.0.0
                - --image-name="ghcr.io/platform-mesh/security-operator"
                - --shutdown-timeout=1m
                - --max-concurrent-reconciles=10
              env:
                - name: FGA_TARGET
                  value: openfga.platform-mesh-system.svc.cluster.local:8081
                - name: KCP_KUBECONFIG
                  value: /api-kubeconfig/kubeconfig
              image: ghcr.io/platform-mesh/security-operator:1.0.0
              imagePullPolicy: IfNotPresent
              livenessProbe:
                failureThreshold: 1
                httpGet:
                  path: /healthz
                  port: 8090
                periodSeconds: 10
              name: RELEASE-NAME-security-operator
              ports:
                - containerPort: 9090
                  name: metrics
                  protocol: TCP
                - containerPort: 8090
                  name: health-port
                  protocol: TCP
              readinessProbe:
                httpGet:
                  path: /readyz
                  port: 8090
                initialDelaySeconds: 5
                periodSeconds: 10
              resources:
                limits:
                  cpu: 260m
                  memory: 512Mi
                requests:
                  cpu: 150m
                  memory: 128Mi
              securityContext:
                readOnlyRootFilesystem: true
                runAsNonRoot: true
                seccompProfile:
                  type: RuntimeDefault
              startupProbe:
                failureThreshold: 30
                httpGet:
                  path: /readyz
                  port: 8090
                periodSeconds: 10
              volumeMounts:
                - mountPath: /api-kubeconfig
                  name: external-api-server
          securityContext:
            runAsNonRoot: true
            seccompProfile:
              type: RuntimeDefault
          serviceAccountName: RELEASE-NAME-security-operator
          terminationGracePeriodSeconds: 10
          volumes:
            - name: external-api-server
              secret:
                secretName: kcp-kubeconfig
  6: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      name: RELEASE-NAME-security-operator-initializer
      namespace: NAMESPACE
    spec:
      revisionHistoryLimit: 3
      selector:
        matchLabels:
          service: RELEASE-NAME-security-operator-initializer
      template:
        metadata:
          labels:
            control-plane: controller-manager
            service: RELEASE-NAME-security-operator-initializer
        spec:
          automountServiceAccountToken: true
          containers:
            - args:
                - initializer
                - --core-module-path=/data/coreModule.fga
                - --test=myval
                - --test2=myval
                - --leader-elect
                - --metrics-bind-address=:9090
                - --health-probe-bind-address=:8090
                - --log-level=warn
                - --region=local
                - --environment=local
                - --image-tag=1.0.0
                - --image-name="ghcr.io/platform-mesh/security-operator"
                - --shutdown-timeout=1m
                - --max-concurrent-reconciles=10
              env:
                - name: FGA_TARGET
                  value: openfga.platform-mesh-system.svc.cluster.local:8081
                - name: KCP_KUBECONFIG
                  value: /api-kubeconfig/kubeconfig
              image: ghcr.io/platform-mesh/security-operator:1.0.0
              imagePullPolicy: IfNotPresent
              livenessProbe:
                failureThreshold: 1
                httpGet:
                  path: /healthz
                  port: 8090
                periodSeconds: 10
              name: RELEASE-NAME-security-operator
              ports:
                - containerPort: 9090
                  name: metrics
                  protocol: TCP
                - containerPort: 8090
                  name: health-port
                  protocol: TCP
              readinessProbe:
                httpGet:
                  path: /readyz
                  port: 8090
                initialDelaySeconds: 5
                periodSeconds: 10
              resources:
                limits:
                  cpu: 260m
                  memory: 512Mi
                requests:
                  cpu: 150m
                  memory: 128Mi
              securityContext:
                readOnlyRootFilesystem: true
                runAsNonRoot: true
                seccompProfile:
                  type: RuntimeDefault
              startupProbe:
                failureThreshold: 30
                httpGet:
                  path: /readyz
                  port: 8090
                periodSeconds: 10
              volumeMounts:
                - mountPath: /api-kubeconfig
                  name: external-api-server
                - mountPath: /data
                  name: core-module
          securityContext:
            runAsNonRoot: true
            seccompProfile:
              type: RuntimeDefault
          serviceAccountName: RELEASE-NAME-security-operator
          terminationGracePeriodSeconds: 10
          volumes:
            - name: external-api-server
              secret:
                secretName: kcp-initializer-kubeconfig
            - configMap:
                name: RELEASE-NAME-security-operator-core-module
              name: core-module
  7: |
    apiVersion: v1
    kind: ServiceAccount
    metadata:
      name: RELEASE-NAME-security-operator
      namespace: NAMESPACE
operator match the snapshot w. sentry:
  1: |
    apiVersion: rbac.authorization.k8s.io/v1
    kind: ClusterRole
    metadata:
      name: RELEASE-NAME-security-operator
    rules:
      - apiGroups:
          - coordination.k8s.io
        resources:
          - leases
        verbs:
          - create
          - delete
          - get
          - list
          - patch
          - update
          - watch
      - apiGroups:
          - ""
        resources:
          - events
          - secrets
        verbs:
          - '*'
      - apiGroups:
          - helm.toolkit.fluxcd.io
        resources:
          - helmreleases
        verbs:
          - get
          - list
          - watch
          - create
          - delete
          - update
          - patch
      - apiGroups:
          - source.toolkit.fluxcd.io
        resources:
          - ocirepositories
          - helmreleases
        verbs:
          - get
          - list
          - create
          - update
          - patch
          - delete
          - watch
  2: |
    apiVersion: rbac.authorization.k8s.io/v1
    kind: ClusterRoleBinding
    metadata:
      name: RELEASE-NAME-security-operator
    roleRef:
      apiGroup: rbac.authorization.k8s.io
      kind: ClusterRole
      name: RELEASE-NAME-security-operator
    subjects:
      - kind: ServiceAccount
        name: RELEASE-NAME-security-operator
        namespace: NAMESPACE
  3: |
    apiVersion: v1
    data:
      coreModule.fga: |2-
        module core

        type user

        type role
          relations
            define assignee: [user,user:*]

        type core_platform-mesh_io_account
          relations
            define parent: [core_platform-mesh_io_account]

            define owner: [role#assignee] or owner from parent
            define member: [role#assignee] or owner or member from parent

            define get: member
            define update: member
            define patch: member
            define delete: owner

            define create_core_platform-mesh_io_accounts: member
            define list_core_platform-mesh_io_accounts: member
            define watch_core_platform-mesh_io_accounts: member

            # org and account specific
            define watch: member

            define create_core_platform-mesh_io_accountinfos: member
            define list_core_platform-mesh_io_accountinfos: member
            define watch_core_platform-mesh_io_accountinfos: member

            define create_apis_kcp_io_apibindings: owner
            define list__apis_kcp_io_apibindings: member
            define watch_apis_kcp_io_apibindings: member

            # IAM specific
            define manage_iam_roles: owner
            define get_iam_roles: member
            define get_iam_users: member

        type core_platform-mesh_io_accountinfo
          relations
            define parent: [core_platform-mesh_io_account]

            define member: member from parent
            define owner: owner from parent

            define get: member
            define watch: member

            # IAM specific
            define manage_iam_roles: owner
            define get_iam_roles: member
            define get_iam_users: member

        type apis_kcp_io_apibinding
          relations
            define parent: [core_platform-mesh_io_account]

            define member: member from parent
            define owner: owner from parent

            define get: member
            define watch: member
            define update: owner
            define patch: owner
            define delete: owner

            # IAM specific
            define manage_iam_roles: owner
            define get_iam_roles: member
            define get_iam_users: member
    kind: ConfigMap
    metadata:
      name: RELEASE-NAME-security-operator-core-module
  4: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      name: RELEASE-NAME-security-operator
      namespace: NAMESPACE
    spec:
      revisionHistoryLimit: 3
      selector:
        matchLabels:
          app: RELEASE-NAME-security-operator
      template:
        metadata:
          labels:
            app: RELEASE-NAME-security-operator
            control-plane: controller-manager
        spec:
          automountServiceAccountToken: true
          containers:
            - args:
                - fga
                - --invite-keycloak-password-file=/keycloak-credentials/secret
                - --leader-elect
                - --metrics-bind-address=:9090
                - --health-probe-bind-address=:8090
                - --log-level=warn
                - --region=local
                - --environment=local
                - --image-tag=1.0.0
                - --image-name="ghcr.io/platform-mesh/security-operator"
                - --shutdown-timeout=1m
                - --max-concurrent-reconciles=10
              env:
                - name: SENTRY_DSN
                  valueFrom:
                    secretKeyRef:
                      key: dsn
                      name: RELEASE-NAME-security-operator-sentry
                - name: TEST_ENV
                  value: test
                - name: FGA_TARGET
                  value: openfga.platform-mesh-system.svc.cluster.local:8081
              image: ghcr.io/platform-mesh/security-operator:1.0.0
              imagePullPolicy: IfNotPresent
              livenessProbe:
                failureThreshold: 1
                httpGet:
                  path: /healthz
                  port: 8090
                periodSeconds: 10
              name: RELEASE-NAME-security-operator
              ports:
                - containerPort: 9090
                  name: metrics
                  protocol: TCP
                - containerPort: 8090
                  name: health-port
                  protocol: TCP
              readinessProbe:
                httpGet:
                  path: /readyz
                  port: 8090
                initialDelaySeconds: 5
                periodSeconds: 10
              resources:
                limits:
                  cpu: 260m
                  memory: 512Mi
                requests:
                  cpu: 150m
                  memory: 128Mi
              securityContext:
                readOnlyRootFilesystem: true
                runAsNonRoot: true
                seccompProfile:
                  type: RuntimeDefault
              startupProbe:
                failureThreshold: 30
                httpGet:
                  path: /readyz
                  port: 8090
                periodSeconds: 10
              volumeMounts:
                - mountPath: /keycloak-credentials
                  name: keycloak-credentials
                  readOnly: true
                - mountPath: /etc/kcp-kubeconfig
                  name: kcp-kubeconfig
                  readOnly: true
          securityContext:
            runAsNonRoot: true
            seccompProfile:
              type: RuntimeDefault
          serviceAccountName: RELEASE-NAME-security-operator
          terminationGracePeriodSeconds: 10
          volumes:
            - name: keycloak-credentials
              secret:
                secretName: keycloak-admin
            - name: kcp-kubeconfig
              secret:
                secretName: kcp-kubeconfig
  5: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      name: RELEASE-NAME-security-operator-generator
      namespace: NAMESPACE
    spec:
      revisionHistoryLimit: 3
      selector:
        matchLabels:
          service: RELEASE-NAME-security-operator-generator
      template:
        metadata:
          labels:
            control-plane: controller-manager
            service: RELEASE-NAME-security-operator-generator
        spec:
          automountServiceAccountToken: true
          containers:
            - args:
                - model-generator
                - --leader-elect
                - --metrics-bind-address=:9090
                - --health-probe-bind-address=:8090
                - --log-level=warn
                - --region=local
                - --environment=local
                - --image-tag=1.0.0
                - --image-name="ghcr.io/platform-mesh/security-operator"
                - --shutdown-timeout=1m
                - --max-concurrent-reconciles=10
              env:
                - name: SENTRY_DSN
                  valueFrom:
                    secretKeyRef:
                      key: dsn
                      name: RELEASE-NAME-security-operator-sentry
                - name: TEST_ENV
                  value: test
                - name: FGA_TARGET
                  value: openfga.platform-mesh-system.svc.cluster.local:8081
              image: ghcr.io/platform-mesh/security-operator:1.0.0
              imagePullPolicy: IfNotPresent
              livenessProbe:
                failureThreshold: 1
                httpGet:
                  path: /healthz
                  port: 8090
                periodSeconds: 10
              name: RELEASE-NAME-security-operator
              ports:
                - containerPort: 9090
                  name: metrics
                  protocol: TCP
                - containerPort: 8090
                  name: health-port
                  protocol: TCP
              readinessProbe:
                httpGet:
                  path: /readyz
                  port: 8090
                initialDelaySeconds: 5
                periodSeconds: 10
              resources:
                limits:
                  cpu: 260m
                  memory: 512Mi
                requests:
                  cpu: 150m
                  memory: 128Mi
              securityContext:
                readOnlyRootFilesystem: true
                runAsNonRoot: true
                seccompProfile:
                  type: RuntimeDefault
              startupProbe:
                failureThreshold: 30
                httpGet:
                  path: /readyz
                  port: 8090
                periodSeconds: 10
              volumeMounts:
                - mountPath: /etc/kcp-kubeconfig
                  name: kcp-kubeconfig
                  readOnly: true
          securityContext:
            runAsNonRoot: true
            seccompProfile:
              type: RuntimeDefault
          serviceAccountName: RELEASE-NAME-security-operator
          terminationGracePeriodSeconds: 10
          volumes:
            - name: kcp-kubeconfig
              secret:
                secretName: kcp-kubeconfig
  6: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      name: RELEASE-NAME-security-operator-initializer
      namespace: NAMESPACE
    spec:
      revisionHistoryLimit: 3
      selector:
        matchLabels:
          service: RELEASE-NAME-security-operator-initializer
      template:
        metadata:
          labels:
            control-plane: controller-manager
            service: RELEASE-NAME-security-operator-initializer
        spec:
          automountServiceAccountToken: true
          containers:
            - args:
                - initializer
                - --core-module-path=/data/coreModule.fga
                - --leader-elect
                - --metrics-bind-address=:9090
                - --health-probe-bind-address=:8090
                - --log-level=warn
                - --region=local
                - --environment=local
                - --image-tag=1.0.0
                - --image-name="ghcr.io/platform-mesh/security-operator"
                - --shutdown-timeout=1m
                - --max-concurrent-reconciles=10
              env:
                - name: SENTRY_DSN
                  valueFrom:
                    secretKeyRef:
                      key: dsn
                      name: RELEASE-NAME-security-operator-sentry
                - name: TEST_ENV
                  value: test
                - name: FGA_TARGET
                  value: openfga.platform-mesh-system.svc.cluster.local:8081
              image: ghcr.io/platform-mesh/security-operator:1.0.0
              imagePullPolicy: IfNotPresent
              livenessProbe:
                failureThreshold: 1
                httpGet:
                  path: /healthz
                  port: 8090
                periodSeconds: 10
              name: RELEASE-NAME-security-operator
              ports:
                - containerPort: 9090
                  name: metrics
                  protocol: TCP
                - containerPort: 8090
                  name: health-port
                  protocol: TCP
              readinessProbe:
                httpGet:
                  path: /readyz
                  port: 8090
                initialDelaySeconds: 5
                periodSeconds: 10
              resources:
                limits:
                  cpu: 260m
                  memory: 512Mi
                requests:
                  cpu: 150m
                  memory: 128Mi
              securityContext:
                readOnlyRootFilesystem: true
                runAsNonRoot: true
                seccompProfile:
                  type: RuntimeDefault
              startupProbe:
                failureThreshold: 30
                httpGet:
                  path: /readyz
                  port: 8090
                periodSeconds: 10
              volumeMounts:
                - mountPath: /etc/kcp-kubeconfig
                  name: kcp-kubeconfig
                  readOnly: true
                - mountPath: /data
                  name: core-module
          securityContext:
            runAsNonRoot: true
            seccompProfile:
              type: RuntimeDefault
          serviceAccountName: RELEASE-NAME-security-operator
          terminationGracePeriodSeconds: 10
          volumes:
            - name: kcp-kubeconfig
              secret:
                secretName: kcp-kubeconfig
            - configMap:
                name: RELEASE-NAME-security-operator-core-module
              name: core-module
  7: |
    apiVersion: external-secrets.io/v1beta1
    kind: ExternalSecret
    metadata:
      name: RELEASE-NAME-security-operator-sentry
      namespace: NAMESPACE
    spec:
      data:
        - remoteRef:
            conversionStrategy: Default
            key: sentry/sentry-dsn
            property: RELEASE-NAME-security-operator-sentry
          secretKey: dsn
      refreshInterval: 10m
      secretStoreRef:
        kind: SecretStore
        name: environment-store
      target:
        creationPolicy: Owner
        deletionPolicy: Retain
        name: RELEASE-NAME-security-operator-sentry
  8: |
    apiVersion: v1
    kind: ServiceAccount
    metadata:
      name: RELEASE-NAME-security-operator
      namespace: NAMESPACE
